src/Uno.Toolkit.RuntimeTests/Tests/StyleExtensionsTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.Toolkit.RuntimeTests.Helpers;
+using Uno.Toolkit.UI;
+using Uno.UI.RuntimeTests;
+
+#if IS_WINUI
+using Microsoft.UI;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Media.Imaging;
+#else
+using Windows.UI;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Imaging;
+#endif
+
+namespace Uno.Toolkit.RuntimeTests.Tests;
+
+[TestClass]
+[RunsOnUIThread]
+internal class StyleExtensionsTest
+{
+	[TestMethod]
+	public async Task StyleExtensionsResourcesPropertyAppliedToStyleTest()
+	{
+		// Arrange
+		var colorBrush = new SolidColorBrush(Colors.DarkGreen);
+		var testKey = "TestKey";
+
+		var resourceDictionary = new ResourceDictionary
+		{
+			{ testKey, colorBrush }
+		};
+
+		var style = new Style(typeof(Button))
+		{
+			Setters =
+			{
+				new Setter { Property = StyleExtensions.ResourcesProperty, Value = resourceDictionary }
```

Comment:
We should also add a test where we simply set StyleExtensions.Resources property directly on the Button, without a Style.